src/backend/distributed/operations/shard_transfer.c

```diff
@@ -413,6 +416,45 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * Insert deferred cleanup records.
+ * The shards will be dropped by background cleaner later.
+ */
+void
+InsertDeferredDropCleanupRecordsForShards(List *shardIntervalList)
+{
+	ListCell *shardIntervalCell = NULL;
+
+	foreach(shardIntervalCell, shardIntervalList)
+	{
+		ShardInterval *shardInterval = (ShardInterval *) lfirst(shardIntervalCell);
+		ListCell *shardPlacementCell = NULL;
+		uint64 oldShardId = shardInterval->shardId;
+
+		/* mark for deferred drop */
+		List *shardPlacementList = ActiveShardPlacementList(oldShardId);
```

Comment:
I think this function should take the group id of the node from which shard is moved. Currently if replication_factor is more than 1 for the shard, it means that a shard move would remove both active placements.

Reply:
just introduced a new function that does the same thing but only for the placements on a certain node